docs/getting_started_windows_vulkan.md

```diff
@@ -0,0 +1,130 @@
+the [Getting Started on Windows with CMake](getting_started_windows_cmake.md)
+guide for instructions.
+
+You must have a physical GPU with drivers supporting Vulkan. We support using
+[SwiftShader](https://swiftshader.googlesource.com/SwiftShader/) (a high
+performance CPU-based implementation of Vulkan) on platforms where the
+[Vulkan-ExtensionLayer](https://github.com/KhronosGroup/Vulkan-ExtensionLayer)
+project builds, but it does not currently
+[build on Windows](https://github.com/KhronosGroup/Vulkan-ExtensionLayer/issues/16).
+
+Vulkan API version > 1.2 is recommended, though older versions with support for
+the `VK_KHR_timeline_semaphore` extension may also work.
+
+## Vulkan Setup
+
+### Background
+
+Vulkan applications interface with Vulkan "drivers", "layers", and "extensions"
+through the Vulkan loader. See LunarG's
+[Architecture of the Vulkan Loader Interfaces](https://vulkan.lunarg.com/doc/view/latest/windows/loader_and_layer_interface.html) page for more information.
+
+### Quick Start
+
+The
+[dynamic_symbols_test](https://github.com/google/iree/blob/master/iree/hal/vulkan/dynamic_symbols_test.cc)
+checks if the Vulkan loader and a valid ICD are accessible.
+
+To configure CMake then build and run the test:
+
+```shell
+$ cmake -B build\ .
+$ cmake --build build\ -j 8 --target iree_hal_vulkan_dynamic_symbols_test
+$ .\build\iree\hal\vulkan\Debug\iree_hal_vulkan_dynamic_symbols_test.exe
+```
+
+Tests in IREE's HAL "Conformence Test Suite" (CTS) actually exercise the Vulkan
+HAL, which includes checking for supported layers and extensions.
+
+Run the
+[allocator test](https://github.com/google/iree/blob/master/iree/hal/cts/allocator_test.cc):
+
+```shell
+$ cmake --build build\ -j 8 --target iree_hal_cts_allocator_test
+$ .\build\iree\hal\cts\Debug\iree_hal_cts_allocator_test.exe
+```
+
+If these tests pass, you can skip down to the next section.
+
+### Setting up the Vulkan Loader
+
+If you see failures to find `vulkan-1.dll` (the Vulkan loader), install it by
+either:
+
+*   Updating your system's GPU drivers
+*   Installing the [Vulkan SDK](https://www.lunarg.com/vulkan-sdk/)
+*   Building the Vulkan loader
+    [from source](https://github.com/KhronosGroup/Vulkan-Loader)
+
+<!--
+### Setting up SwiftShader
+
+TODO(scotttodd): Document when SwiftShader supports `VK_KHR_timeline_semaphore`
+                 Or Vulkan-ExtensionLayer builds for Windows
+
+### Setting up Vulkan-ExtensionLayer
+
+TODO(scotttodd): Document when Vulkan-ExtensionLayer builds for Windows
+-->
+
+## Using IREE's Vulkan Compiler Target and Runtime Driver
+
+### Compiling for the Vulkan HAL
+
+Pass the flag `-iree-hal-target-backends=vulkan-spirv` to `iree-translate.exe`:
+
+```shell
+$ cmake --build build\ -j 8 --target iree_tools_iree-translate
+$ .\build\iree\tools\Debug\iree-translate.exe -iree-mlir-to-vm-bytecode-module -iree-hal-target-backends=vulkan-spirv .\iree\tools\test\simple.mlir -o .\build\module.fb -print-ir-after-all
```

Comment:
print-ir-after-all is not necessary i think.

Reply:
The command has no output without some "print" flag, so I thought it would be useful to see some output. I moved that flag to a tip.